src/shared/lib/utils/survey.ts

```diff
@@ -13,4 +15,208 @@ export const invertColor = (hex: string) => {
   return yiqColorSpaceValue >= 128 ? colors.darkerGrey : colors.white;
 };
 
-export const replaceTextWithScreenVariables = (string: string) => string; // @todo complete this function
+export const replaceTextWithScreenVariables = (markdown: string): string => {
+  const activityItems: ActivityItems = []; //@todo get these from some storage?
+  const answers: Answer[] = [];
+  const replacer = new MarkdownVariableReplacer(
+    markdown,
+    activityItems,
+    answers,
+  );
+  return replacer.process();
+};
+
+type TimeRangeAnswer = {
+  from: {
+    hour: number;
+    minute: number;
+  };
+  to: {
+    hour: number;
+    minute: number;
+  };
+};
+
+type TimeRangeInput = {
+  inputType: 'timeRange';
+  answer: TimeRangeAnswer;
+};
+
+type DateAnswer = {
+  year: number;
+  month: number;
+  day: number;
+};
+
+type DateInput = {
+  inputType: 'date';
+  answer: DateAnswer;
+};
+
+type TextInput = {
+  inputType: 'text';
+  answer: string;
+};
+
+type RadioInput = {
+  inputType: 'radio';
+  answer: number;
+};
+
+type AgeSelectorInput = {
+  inputType: 'ageSelector';
+  answer: number;
+};
+
+type SliderInput = {
+  inputType: 'slider';
+  answer: number;
+};
+
+type CheckBoxInput = {
+  inputType: 'checkbox';
+  answer: string[];
+};
+
+type Answer = DateAnswer | TimeRangeAnswer | string | string[] | number;
+
+type InputAnswer =
+  | DateInput
+  | TimeRangeInput
+  | TextInput
+  | RadioInput
+  | CheckBoxInput
+  | AgeSelectorInput
+  | SliderInput;
+
+type ActivityItems = {
+  variableName: string;
+  inputType: InputAnswer['inputType'];
+  valueConstraints: {
+    itemList: { name: string; value: string | number }[];
+  };
+}[];
+
+export class MarkdownVariableReplacer {
```

Comment:
Please put it in a separate file.
You can create the "src/shared/survey" folder and move there everything that you created here.